bezier_library/src/bezier_library.cpp

```diff
@@ -1112,80 +1112,102 @@ bool Bezier::generateTrajectory(
-    start_point.y = end_point.y - length * way_points_vector[k].linear().col(2)[1];
-    start_point.z = end_point.z - length * way_points_vector[k].linear().col(2)[2];
-
-    marker.points.push_back(start_point);
-    marker.points.push_back(end_point);
 
-    if (points_color_viz[k] == true)
+    if(boost::iequals(where,"path"))
     {
-      marker.color.r = 0.0f;
-      marker.color.g = 1.0f;
-      marker.color.b = 0.0f;
+      if(points_color_viz[i])
+        visual_tools_->publishZArrow(way_points_vector[i],
+                                     rviz_visual_tools::GREEN,
+                                     rviz_visual_tools::XSMALL, length, i);
     }
-    else
+    else if(boost::iequals(where,"extrication"))
     {
-      marker.color.r = 1.0f;
-      marker.color.g = 0.0f;
-      marker.color.b = 0.0f;
+      if(!points_color_viz[i])
+        visual_tools_->publishZArrow(way_points_vector[i],
+                                     rviz_visual_tools::RED,
+                                     rviz_visual_tools::XSMALL, length, i);
     }
-    marker.color.a = 0.7;
-    marker.lifetime = ros::Duration();
-    markers.markers.push_back(marker);
+    else
+      visual_tools_->publishZArrow(way_points_vector[i],
+                                   (points_color_viz[i] ? rviz_visual_tools::GREEN : rviz_visual_tools::RED),
+                                   rviz_visual_tools::XSMALL, length, i);
   }
-  while (normal_publisher.getNumSubscribers() < 1)
+
+  visual_tools_->triggerBatchPublish();
+  ros::spinOnce();
+
+}
+
+void Bezier::RVIZdisplayAxes(
+    const std::vector<Eigen::Affine3d, Eigen::aligned_allocator<Eigen::Affine3d> > &way_points_vector,
+    const std::vector<bool> points_color_viz,
+    const std::string name,
+    const ros::Publisher &axles_publisher,
+    const std::string where)
+{
+
+  if (way_points_vector.size() != points_color_viz.size())
   {
-    ROS_WARN_ONCE("Bezier::displayNormal: Please create a subscriber to the normals marker");
-    sleep(1);
+    ROS_ERROR_STREAM("Bezier::RVIZdisplayAxles: Path vector and bool vector have different sizes");
+    return;
+  }
+
+  visual_tools_.reset(new rviz_visual_tools::RvizVisualTools("base_link",name));
+  visual_tools_->waitForSubscriber(axles_publisher, 0.5, true);
+  visual_tools_->setLifetime(0);
+  visual_tools_->enableBatchPublishing();
+
+  double length = 0.01;
+  double radius = 0.0005;
+  for (unsigned int i = 0; i < way_points_vector.size(); i++)
+  {
+    if(boost::iequals(where,"path"))
+    {
+      if(points_color_viz[i] && (i%2 == 0))
+        visual_tools_->publishAxis(way_points_vector[i],length,radius);
+    }
+    else if(boost::iequals(where,"extrication") && (i%2 == 0))
+    {
+      if(!points_color_viz[i])
+        visual_tools_->publishAxis(way_points_vector[i],length,radius);
+    }
+    else
+    {
+      if(i%2 == 0)
+        visual_tools_->publishAxis(way_points_vector[i],length,radius);
```

Comment:
Put modulo 2 as a parameter so that the user can choose to display all of the axis or only 1 of 2... one of 4 ...